### PR TITLE
DOC: stats: complete references and links for descriptive stats

### PR DESCRIPTION
**Pull request by qbarthelemy**
#### Reference issue

None. No code modification, only documentation.

#### What does this implement/fix?

This PR completes references and cross-links for descriptive stats functions.

### CODE REVIEW

File: scipy/stats/_stats_py.py [hunk 1]
```diff
@@ -384,6 +386,8 @@ def pmean(a, p, *, axis=0, dtype=None, weights=None):
 
         \left( \frac{ 1 }{ n } \sum_{i=1}^n a_i^p \right)^{ 1 / p }  \, .
 
+    When p=0, it returns the geometric mean.
```
Comment (tupui) [inline scipy/stats/_stats_py.py:389]:
```suggestion
    When ``p=0``, it returns the geometric mean.
```

File: scipy/stats/_stats_py.py [hunk 2]
```diff
@@ -2671,12 +2675,25 @@ def zscore(a, axis=0, ddof=0, nan_policy='propagate'):
         The z-scores, standardized by mean and standard deviation of
         input array `a`.
 
+    See Also
+    --------
+    numpy.mean : Arithmetic average
+    numpy.std : Arithmetic standard deviation
+    gzscore : Geometric standard score
```
Comment (tupui) [inline scipy/stats/_stats_py.py:2682]: This is making the CI fail. I am not sure why though. Could you check this locally?

**Review by tupui — CHANGES_REQUESTED** [review]
Thanks @qbarthelemy, just a nitpick and also the CI failure is related.

File: scipy/stats/_stats_py.py (hunk 2 above)
Comment (WarrenWeckesser) [reply scipy/stats/_stats_py.py:2682]: FWIW: I can reproduce the warning locally, running either Sphinx 4.5.0 or 5.3.0.  As an experiment, I tried adding `gzscore` to the `See Also` section of a bunch of other functions, and I got the warning from `gmean`, `hmean` and `zscore`, but not from `pmean`, `tmean`, `mode` and others.  Go figure.  Do we have any resident Sphinx gurus who can take a look at this?

Comment (tupui) [reply scipy/stats/_stats_py.py:2682]: Might be an order thing if the function is defined before or after? Does it fix it if we instead use a full resolution with `scipy.stats.gzscore`?

Comment (qbarthelemy) [reply scipy/stats/_stats_py.py:2682]: Doc is OK when building `scipy.stats.zscore`
https://output.circle-artifacts.com/output/job/0b43ce1a-05db-4959-a5d5-9101dd342737/artifacts/0/html/reference/generated/scipy.stats.zscore.html

Doc is KO when building `scipy.stats.mstats.zscore`
https://output.circle-artifacts.com/output/job/0b43ce1a-05db-4959-a5d5-9101dd342737/artifacts/0/html/reference/generated/scipy.stats.mstats.zscore.html

Comment (tupui) [reply scipy/stats/_stats_py.py:2682]: Let's use `scipy.stats.zscore` as we in the end want to slowly move away from `mstats`.

**Comment by tupui** [comment]
p.s. please don't forget to add some skip instructions for the CI 😃 (`[skip actions] [skip circle]`, we need Azure and CircleCI for doc PRs).

**Comment by qbarthelemy** [comment]
Sorry, but I don’t understand what changes I need to make to solve the current warning.

File: scipy/stats/_stats_py.py (hunk 2 above)
Comment (tupui) [inline scipy/stats/_stats_py.py:2682]:
```suggestion
    scipy.stats.zscore : Geometric standard score
```

File: scipy/stats/_stats_py.py [hunk 3]
```diff
@@ -2913,14 +2935,15 @@ def gstd(a, axis=0, ddof=1):
...
-    ndarray or float
+    gstd : ndarray or float
         An array of the geometric standard deviation. If `axis` is None or `a`
         is a 1d array a float is returned.
 
     See Also
     --------
     gmean : Geometric mean
     numpy.std : Standard deviation
+    gzscore : Geometric standard score
```
Comment (tupui) [inline scipy/stats/_stats_py.py:2946]:
```suggestion
    scipy.stats.zscore : Geometric standard score
```

**Review by tupui — COMMENTED** [review]
This I think

**Comment by tupui** [comment]
Ooops can you re commit without the skip circle? My bad I meant to do `skip cirrus`.

**Review by tupui — APPROVED** [review]
Alright CI is green now. Thank you for pushing this @qbarthelemy

**Comment by WarrenWeckesser** [comment]
So we don't forget, I created an issue about the Sphinx warning: https://github.com/scipy/scipy/issues/17738

**Pull request merged**